src/ucp/proto/proto_common.c

```diff
@@ -661,9 +650,7 @@ void ucp_proto_request_zcopy_completion(uct_completion_t *self)
 
 int ucp_proto_is_short_supported(const ucp_proto_select_param_t *select_param)
 {
-    /* Short protocol requires contig/host */
-    return (select_param->dt_class == UCP_DATATYPE_CONTIG) &&
-           UCP_MEM_IS_HOST(select_param->mem_type);
+    return (select_param->dt_class == UCP_DATATYPE_CONTIG);
```

Comment:
As I understand, we can enable it for iov, but generic requires calling custom pack/unpack (as bcopy do), and we do not support generic even for zcopy.

BTW eager/singe/zcopy also disabled for IOV, should we enable it for empty messages?

Reply:
Discussed offline.

I think it could be done in the separate PR. I want to propose the following solution (https://github.com/ivankochin/ucx/pull/5) which would be common for all the protocols.